packages/react-components/src/utils/common.ts

```diff
@@ -75,3 +76,66 @@ export function equals(a: Array<string>, b: Array<string>): boolean {
     a.length === b.length && a.every((element, index) => element === b[index])
   );
 }
+
+export function splitListBy<T>(
```

Comment:
I think this function is legit. We do want to split lists based on the the presence of a key being there or not. Could be optimized into a single reduce that returns a tupple. I think there is a better way to type this generically using something like whats in packages/dom-test-utils/src/css.ts except instead of css object it uses the generic passed in if that will work